excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/HiddenFieldsValidator.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.ccd.definition.store.excel.validation;
+
+import uk.gov.hmcts.ccd.definition.store.excel.endpoint.exception.MapperException;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionDataItem;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionSheet;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.ColumnName;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.SheetName;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class HiddenFieldsValidator {
+    Boolean retainHiddenValue;
+
+    public Boolean parseHiddenFields(DefinitionDataItem definitionDataItem, Map<String, DefinitionSheet> definitionSheets) {
+        final DefinitionSheet caseEventToFields = definitionSheets.get(SheetName.CASE_EVENT_TO_FIELDS.getName());
+        final DefinitionSheet caseFields = definitionSheets.get(SheetName.CASE_FIELD.getName());
+        Optional<DefinitionDataItem> caseField =
+            caseFields.getDataItems().stream().filter(caseFieldDataItem ->
+                definitionDataItem.getId().equals(caseFieldDataItem.getString(ColumnName.FIELD_TYPE))).findFirst();
+        caseField.ifPresent(ddi -> {
+            Optional<DefinitionDataItem> caseEventToField = caseEventToFields.getDataItems()
+                .stream().filter(definitionDataItem1 -> ddi.getId().equals(definitionDataItem1.getCaseFieldId())).findFirst();
+            caseEventToField.ifPresent(caseEventToFieldDataItem -> {
+                String fieldShowCondition = caseEventToFieldDataItem.getString(ColumnName.FIELD_SHOW_CONDITION);
+                Boolean caseFieldRetainHiddenValue = caseEventToFieldDataItem.getBoolean(ColumnName.RETAIN_HIDDEN_VALUE);
+                if (fieldShowCondition == null && Boolean.TRUE.equals(definitionDataItem.getBoolean(ColumnName.RETAIN_HIDDEN_VALUE))) {
```

Comment:
Might be useful to extract each of the `if` checks to separate methods with appropriate names for readability, rather than the comments